src/kOS/Communication/CommNetConnectivityManager.cs

```diff
@@ -105,15 +105,15 @@ public bool HasConnection(Vessel vessel1, Vessel vessel2)
                 return true;
 
             // We next need to query the network to find a connection between the two vessels.
-            // I found no exposed method for accessing chached paths directly, other than the
+            // I found no exposed method for accessing cached paths directly, other than the
             // control path.
 
             // WARNING: In stock this will only work for vessels with a relay antenna installed.
             // This is a limitation put in place to improve performance in the stock game, and
             // there isn't a very good way around it.
             var net = CommNetNetwork.Instance.CommNet;
             tempPath = new CommPath();
-            return net.FindPath(vessel1.Connection.Comm, tempPath, vessel2.Connection.Comm) || net.FindPath(vessel2.Connection.Comm, tempPath, vessel1.Connection.Comm);
+            return vessel1 == vessel2 || net.FindPath(vessel1.Connection.Comm, tempPath, vessel2.Connection.Comm) || net.FindPath(vessel2.Connection.Comm, tempPath, vessel1.Connection.Comm);
```

Comment:
That's a fair point.  The reason we use the vessel id comparison is that the default equality comparison walks the fields on the object to check equality, which compares ID, and Name, and all the underlying unity fields.  Which means that comparing a single GUID should be faster than comparing all of those fields.